assets/js/components/ViewOnlyMenu/index.js

```diff
@@ -42,13 +44,18 @@ import Tracking from './Tracking';
 export default function ViewOnlyMenu() {
 	const [ menuOpen, setMenuOpen ] = useState( false );
 	const menuWrapperRef = useRef();
+	const viewContext = useViewContext();
 
 	useClickAway( menuWrapperRef, () => setMenuOpen( false ) );
 	useKeyCodesInside( [ ESCAPE, TAB ], menuWrapperRef, () =>
 		setMenuOpen( false )
 	);
 
 	const toggleMenu = useCallback( () => {
+		if ( ! menuOpen ) {
+			trackEvent( `${ viewContext }_headerbar`, 'open_viewonly' );
+		}
+
 		setMenuOpen( ! menuOpen );
 	}, [ menuOpen ] );
```

Comment:
Can we add viewContext to the dependency list?

```suggestion
	}, [ menuOpen, viewContext ] );
```